src/propmcwf.jl

```diff
@@ -0,0 +1,94 @@
+immutable QuMCWF <: QuPropagatorMethod
+    options::Dict{Symbol, Any}
+end
+
+QuMCWF() = QuMCWF(Dict())
+
+type QuMCWFEnsemble{QA<:QuBase.AbstractQuArray}
+    rho::QA
+    ntraj::Int
+    decomp
+    QuMCWFEnsemble(rho, ntraj, decomp) = new(rho, ntraj, decomp)
+end
+
+QuMCWFEnsemble{QM<:QuBase.AbstractQuMatrix}(rho::QM, ntraj=500) = QuMCWFEnsemble{QM}(rho, ntraj, eigfact(full(coeffs(rho))))
+QuMCWFEnsemble{QV<:QuBase.AbstractQuVector}(psi::QV, ntraj=500) = QuMCWFEnsemble{QV}(psi, ntraj, nothing)
+
+Base.start(mcwfensemble::QuMCWFEnsemble) = 1
+
+function Base.next(mcwfensemble::QuMCWFEnsemble, i::Int)
+    state = draw(mcwfensemble)
+    return (state, i+1)
+end
+
+function Base.done(mcwfensemble::QuMCWFEnsemble, i::Int)
+    i > mcwfensemble.ntraj
+end
+
+function draw{QM<:QuBase.AbstractQuMatrix}(mcwfensemble::QuMCWFEnsemble{QM})
+    r = rand() # draw random number from [0,1)
+    pacc = 0.
+    for i=1:length(mcwfensemble.decomp[:values])
+        pacc = pacc + mcwfensemble.decomp[:values][i]
+        if pacc >= r
+            return QuArray(complex(mcwfensemble.decomp[:vectors][:,i]))
+        end
+    end
+end
+
+function draw{QV<:QuBase.AbstractQuVector}(mcwfensemble::QuMCWFEnsemble{QV})
+    return copy(mcwfensemble.rho)
+end
+
+function eff_hamiltonian(lme::QuLindbladMasterEq)
+    heff = lme.hamiltonian
+    for c_ops in lme.collapse_ops
+       heff = heff - im*0.5*c_ops'*c_ops
+    end
+    return heff
+end
+
+function propagate(prob::QuMCWF, eq::QuLindbladMasterEq, t, current_t, psi)
+    jtol = 1.e-6  # jump tolerance
+    # get information of QME
+    heff = eff_hamiltonian(eq)
+    c_ops = eq.collapse_ops
+    eps = rand() # draw random number from [0,1)
```

Comment:
Actually we need to use the `eps` from the previous step. Only for the first step a new random number has to be generated.